### DISCUSSION TITLE
Add support for TMP117 temperature sensor?

### DISCUSSION THREAD

**Discussion by atifan** [arendst/Tasmota#14139] [Ideas]
Hi.
Would be nice if someone could add support for the TMP117 temperature sensor.
Its a not cheap but very accurate sensor.
Here is the datasheet: https://www.ti.com/lit/ds/symlink/tmp117.pdf?ts=1639991283833&ref_url=https%253A%252F%252Fwww.google.com%252F

Other Firmwares like ESPHome support this sensor, so maybe a driver is given to use? I don´t know :)

**Comment by atifan** [comment]
Looks like there is a driver for TMP117 for tasmota, but how can I use this? Maybe it could be integrated in Tasmota?
https://github.com/TomsTek/tasmota-berry-TMP117-driver/blob/main/tmp117.be

**Reply by barbudor** [reply]
This a driver written in Berry language supported only on ESP32
If you have an ESP32 you need to upload the file to the flash file system

If you don't, you can write a native driver in c++ and contribute to Tasmota with a pull request

Note that there are already a lot of very good temp sensors supported by Tasmota
